src/pvnet_app/app.py

```diff
@@ -126,61 +121,78 @@ def get_boolean_env_var(env_var: str, default: bool) -> bool:
+        num_workers (int): Number of workers to use to load batches of data. When set to default
+            value of -1, it will use one less than the number of CPU cores workers.
 
     This app expects these environmental variables to be available:
         - DB_URL
         - NWP_UKV_ZARR_PATH
         - NWP_ECMWF_ZARR_PATH
         - SATELLITE_ZARR_PATH
     The following are options
-        - PVNET_V2_VERSION, pvnet version, default is a version above
-        - PVNET_V2_SUMMATION_VERSION, the pvnet version, default is above
+        - SENTRY_DSN, optional link to sentry
+        - ENVIRONMENT, the environment this is running in, defaults to local
         - USE_ADJUSTER, option to use adjuster, defaults to true
         - SAVE_GSP_SUM, option to save gsp sum for pvnet_v2, defaults to false
         - RUN_EXTRA_MODELS, option to run extra models, defaults to false
         - DAY_AHEAD_MODEL, option to use day ahead model, defaults to false
-        - SENTRY_DSN, optional link to sentry
-        - ENVIRONMENT, the environment this is running in, defaults to local
         - USE_ECMWF_ONLY, option to use ecmwf only model, defaults to false
         - USE_OCF_DATA_SAMPLER, option to use ocf_data_sampler, defaults to true
-
-    Args:
-        t0 (datetime): Datetime at which forecast is made
-        gsp_ids (array_like): List of gsp_ids to make predictions for. This list of GSPs are summed
-            to national.
-        write_predictions (bool): Whether to write prediction to the database. Else returns as
-            DataArray for local testing.
-        num_workers (int): Number of workers to use to load batches of data. When set to default
-            value of -1, it will use one less than the number of CPU cores workers.
     """
+
+    # ---------------------------------------------------------------------------
+    # 0. Basic set up
+
+    # If inference datetime is None, round down to last 30 minutes
+    if t0 is None:
+        t0 = pd.Timestamp.now(tz="UTC").replace(tzinfo=None).floor("30min")
+    else:
+        t0 = t0.floor("30min")
+
+    assert len(gsp_ids)>0, "No GSP IDs provided"
+
     if num_workers == -1:
         num_workers = os.cpu_count() - 1
     if num_workers > 0:
         # Without this line the dataloader will hang if multiple workers are used
         dask.config.set(scheduler="single-threaded")
 
-    assert len(gsp_ids)>0, "No GSP IDs provided"
-
+    # Unpack the environment variables
     use_day_ahead_model = get_boolean_env_var("DAY_AHEAD_MODEL", default=False)
     use_ecmwf_only = get_boolean_env_var("USE_ECMWF_ONLY", default=False)
     run_extra_models = get_boolean_env_var("RUN_EXTRA_MODELS", default=False)
     use_ocf_data_sampler = get_boolean_env_var("USE_OCF_DATA_SAMPLER", default=True)
     use_adjuster = get_boolean_env_var("USE_ADJUSTER", default=True)
     save_gsp_sum = get_boolean_env_var("SAVE_GSP_SUM", default=False)
+    db_url = os.getenv("DB_URL")
```

Comment:
cam we add something that errors if DB_URL is not set. There's a weird error message if we pass None into creating a connection string, that could be something we do in `DatabaseConnection` instead